### ISSUE TITLE
BitBucket Server support

### ISSUE THREAD

**Issue by czechboy0** [czechboy0/Buildasaur#238]

**Comment by mrgrauel** [comment]
:+1:

**Comment by tamastimar** [comment]
Meaning support for self-hosted BitBucket?

**Comment by czechboy0** [comment]
@tamastimar Yes, that's what is meant by that. We support BitBucket cloud-hosted repositories already.

**Comment by krin-san** [comment]
It's a feature I want to have 'cause it's blocking my corporate Xcode Server + Buildasaur + DangerCI  integration for a PR review process.

I found one of the forks with BitbucketServer support: https://github.com/buildasaurs/Buildasaur/compare/master...gchiong:feature/bitbucket_server_support
Would be nice to have it in a native codebase because this fork should be migrated to the latest master state.

**Comment by krin-san** [comment]
I rebased @gchiong branch with bitbucket server support on the latest `buildasaurs/master` state. It's not a perfect version to make a pull request due to some architecture style conflicts, but it's good enough to try it out.
https://github.com/krin-san/Buildasaur/tree/feature/bitbucket_server_support
